rllib/utils/pre_checks.py

```diff
@@ -0,0 +1,51 @@
+"""Common pre-checks for all RLlib experiments."""
+import logging
+
+from ray.rllib.env import BaseEnv, MultiAgentEnv, RemoteVectorEnv, VectorEnv
+import gym
+
+logger = logging.getLogger(__name__)
+
+
+def check_env(
+        env: [BaseEnv, gym.Env, MultiAgentEnv, RemoteVectorEnv,
+              VectorEnv]) -> None:
+    """Run pre-checks on env that uncover common errors in environments.
+
+    Args:
+        env: Environment to be checked.
+
+    Raises:
+        ValueError: If env is not one of the supported types.
+    """
+    supported_types = (BaseEnv, gym.Env, MultiAgentEnv, RemoteVectorEnv,
```

Comment:
define this as a constant at the top of the file, and use it also in the type annotation?
can we also start a simple unit test file with this PR? basically try to fail whatever tests we are checking here.

Reply:
of course! I will add a constant and add tests here for every one of the checks

Reply:
I ended not being able to add this as a constant in this file because of circular imports. Not sure where else I would put it -- maybe in rllib/env/__init__.py but im not sure if thats necessary. Lmk what you think @gjoliver